### ISSUE TITLE
Missing "luxon" in BOM file

### ISSUE THREAD

**Issue by sresch4b** [CycloneDX/cyclonedx-webpack-plugin#778]
This issue is a successor of the issue #745, where we also found out, that the "luxon" package is missing in the bom file.

For reproduction of this issue, I've updated the test repo:
https://github.com/sresch4b/cyclonedx-issue

**Comment by jkowalleck** [comment]
tried to reproduce: 

et webpack's `state` to `detailed`, logs look like 
```text
LOG from CycloneDxWebpackPlugin/ComponentFetcher/Extractor
    start building Components from modules...
    try to build new Component from PkgPath .../package.json
    try to build new Component from PkgPath .../node_modules/@apollo/client/package.json
    try to build new Component from PkgPath .../node_modules/luxon/src/package.json
    try to build new Component from PkgPath .../node_modules/luxon/src/package.json
```

```SHellSession
$ cat .../node_modules/luxon/src/package.json
{
  "type": "module",
  "version": "3.3.0"
}
```

So the "luxon" had decided to make the `src` folder a package? Well, that is fine, but they lacked some useful information in it.
A fix would be, if "luxon" added a proper name to the package they defined in `luxon/src/package.json`

**Comment by jkowalleck** [comment]
@sresch4b there is actually nothing i can do here.
You might want to ask the "luxon" project to add additional information in their [`src/package.json`](https://github.com/moment/luxon/blob/master/src/package.json).

**Comment by sresch4b** [comment]
@jkowalleck
Ok, thanks for checking this.
I think this is the point where we can stop, because `luxon` could be just the first of an uncountable number of (npm) packages that use a `package.json` in a different way than we might expect.

Thanks for your support and insight into this topic!

**Comment by jkowalleck** [comment]
I am very sorry. 
I wish it would be otherwise.

I thought about adding a rule, that the package without a name is ignored, and that I could just traverse the directory path up until a `package.json` with a name is found. 
But this might lead to different problems, and false positives, and ... .

Better not change the current implementation, unless a proper decision paper was written,  the whole universe (constraints, identities, connections, boundaries) is clear and proven right.

The thing is, that webpack is a bundler/packer, it does not care for packages nor registries nor sources. It takes what is there, and builds based on a reproducible best-effort.
And so does this plugin.

**Comment by jkowalleck** [comment]
should be fixed by #1284
